homeassistant/components/venstar/sensor.py

```diff
@@ -166,6 +172,26 @@ def native_unit_of_measurement(self) -> str:
         ),
         name_fn=lambda coordinator, sensor_name: f"{coordinator.client.name} {sensor_name.replace(' Temp', '')} Temperature",
     ),
+    VenstarSensorEntityDescription(
+        key="co2",
+        device_class=SensorDeviceClass.CO2,
+        state_class=SensorStateClass.MEASUREMENT,
+        uom_fn=lambda coordinator: CONCENTRATION_PARTS_PER_MILLION,
+        value_fn=lambda coordinator, sensor_name: coordinator.client.get_sensor(
+            sensor_name, "co2"
+        ),
+        name_fn=lambda coordinator, sensor_name: f"{coordinator.client.name} {sensor_name} CO2",
+    ),
+    VenstarSensorEntityDescription(
+        key="iaq",
+        device_class=SensorDeviceClass.AQI,
+        state_class=SensorStateClass.MEASUREMENT,
+        uom_fn=lambda coordinator: "",
```

Comment:
I think if you have no units, you should set to `None` (or maybe just omit it)

I see that airvisual has units stated as "AQI".  This seems very similar to the [venstar manual I found here](https://files.venstar.com/thermostats/exploreriaq/documents/T4975-IAQ_UM_011822.pdf).  Although venstar seem very vague about what the units actually are, it could be a good starting point.

Reply:
Using `None` gave the following pre-commit error:

```
homeassistant/components/venstar/sensor.py:189: error: Argument "uom_fn" to "VenstarSensorEntityDescription" has incompatible type "Callable[[Any], None]"; expected "Callable[[Any], str]"  [arg-type]
homeassistant/components/venstar/sensor.py:189: error: Incompatible return value type (got "None", expected "str")  [return-value]
Found 2 errors in 1 file (checked 1 source file)
```

And removing the `uom_fn` argument gave this one:

```
homeassistant/components/venstar/sensor.py:185: error: Missing positional argument "uom_fn" in call to "VenstarSensorEntityDescription"  [call-arg]
Found 1 error in 1 file (checked 1 source file)
```

So I think we definitely _could_ omit it from the required properties of the dataclass mixin but I'm not sure if that means it has to be set as an optional parameter. Specifying a blank string seems more correct as we're explicitly saying there is no unit of measure. I'm fine with either approach, however.


I did see AQI and it seems to be measured on a similarly unitless scale; for example, compare [the table on airnow.gov](https://www.airnow.gov/aqi/aqi-basics/):

![image](https://user-images.githubusercontent.com/32495955/167271926-c17d5c44-7bca-4c3a-ac54-c4c4bf7eabeb.png)

to the one in Venstar's docs:

![image](https://user-images.githubusercontent.com/32495955/167271914-cee492d1-c296-44d2-80f5-59296b77b036.png)

Though I'm not sure appending either "AQI" or "IAQ" as the unit of measurement is correct. Seems more correct that they don't have units at all.

Reply:
When I search for `unit_of_measurement=None` and `unit_of_measurement=""`, there are some integrations that use None and some that use "", so I suppose it is not critical, but there are more that use `None`.

e.g.
https://github.com/home-assistant/core/blob/55cb35046ec3b2e9072d4bebfbd7259ab2938253/homeassistant/components/metoffice/sensor.py#L105

as for replacing with `None`, the mixin will want to call whatever you assign to uom_fn, so you'll need something like
uom_fn=lambda: None

You don't need `coordinator` as a parameter for the lambda function because it's not used.  I think that can be deleted from a few other `uom_fn` entries nearby too.

Reply:
Cool -- `None` it is. The `coordinator` parameter _is_ being used in the `temperature_unit` function so it appears to be added to the other `uom_fn` function args in order to meet the `Callable[[Any], str]` interface.

Reply:
I see... it is strange to pass a parameter that is unused.  Actually it could be anything.

I would be tempted to use `uom_fn=lambda _: None` which I think should work although I am not able to test it.

Reply:
Agreed. Good point; I didn't stop long enough to think of that :upside_down_face: I'm not really sure why this mixin wrapper is even here but didn't want to remove it in case it was actually useful.

Reply:
Yeah, it is kind of an odd thing to see used like this.  I think it is because the original code wants to use the standard `EntityDescription` class but wants to pick up the temperature units from the api.